src/utils.rs

```diff
@@ -1,6 +1,23 @@
 use std::future::Future;
 use std::pin::Pin;
 
+use crate::Error;
+
 /// An owned dynamically typed [`Future`] for use in cases where you can't
 /// statically type your result or need to add some indirection.
 pub(crate) type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+
+/// Create a response from an [`Error`]. The response's status code will be
+/// retrieved from the error and the response body is set to the error message
+/// for non-500 errors.
+pub(crate) fn create_error_response(err: Error) -> http_types::Response {
+    let mut res = http_types::Response::new(err.status());
+    res.set_content_type(http_types::mime::PLAIN);
+    // Only send the message if it is a non-500 range error. All
+    // errors default to 500 by default, so sending the error
+    // body is opt-in at the call site.
+    if !res.status().is_server_error() {
+        res.set_body(err.to_string());
+    }
+    res
+}
```

Comment:
This is the implementation part I am not sure about. I have the feeling that having this as a utility method is not the best solution, so I am definitely open to other ideas of how to streamline the error responses here?